docs/advanced.md

```diff
@@ -221,6 +221,35 @@ with httpx.Client(headers=headers) as client:
     ...
 ```
 
+## Monitoring download progress
+
+If you need to monitor download progress of large responses, you can use stream and the `response.last_raw_chunk_size` property.
+
+For example, showing a progress bar using the [`tqdm`](https://github.com/tqdm/tqdm) library while a response is being downloaded could be done like this…
+
+```python
+import tempfile
+
+import httpx
+from tqdm import tqdm
+
+with tempfile.NamedTemporaryFile() as download_file:
+    url = "https://speed.hetzner.de/100MB.bin"
+    with httpx.stream("GET", url) as response:
+        if "Content-Length" in response.headers:
+            total = int(response.headers["Content-Length"])
+        else:
+            total = None
```

Comment:
I know about @tomchristie's comment here… https://github.com/encode/httpx/pull/1268#discussion_r485710581

But for this particular endpoint we know that we'll get a `Content-Length` in response, so should we simplify this to…?

```suggestion
        total = int(response.headers["Content-Length"])
```

I guess the motivation is to show a general-purpose example, but at the same time it's a bit confusing since _for any particular example_ there's only one branch we'll ever go through.

Reply:
I don't know.

On one hand it's good idea to show that `tqdm` accepts `total=None`. On other hand an extra `if` statement can mislead someone (moreover, it's the other library, not `httpx`).

I'd like to make an example easier, as you proposed.

@tomchristie , what do you think?

Reply:
I'm happy with either...

Keeping it simple:

```python
total = int(response.headers["Content-Length"])
```

Or just being a little more careful...

```python
content_length = response.headers.get("Content-Length")
total = None if content_length is None else int(content_length)
```